src/widgets/unfold_widgets/results_widget.py

```diff
@@ -23,10 +23,18 @@ def load_widget(self):
         for i in reversed(range(self.results_tab_widget.count())):
             self.results_tab_widget.removeTab(i)
 
+        if self.engine.state.last_algorithm is not None:
+            last_technique, last_algorithm = self.engine.state.last_algorithm
+        else:
+            last_technique = last_algorithm = ""
+
         for technique, algorithms in self.engine.state.algorithm_results_widgets.items():
             for algorithm, results in algorithms.items():
                 algorithm_result_tab_widget = QTabWidget()
+                result_idx = 0
                 for i, result_widget in enumerate(results):
-                    algorithm_result_tab_widget.addTab(result_widget, f'{i+1}')
-                self.results_tab_widget.addTab(algorithm_result_tab_widget, f'{technique}: {algorithm}')
-
+                    result_idx = algorithm_result_tab_widget.addTab(result_widget, f'{i+1}')
+                algorithm_idx = self.results_tab_widget.addTab(algorithm_result_tab_widget, f'{technique}: {algorithm}')
+                if technique == last_technique and algorithm == last_algorithm:
+                    self.results_tab_widget.setCurrentIndex(algorithm_idx)
+                    self.results_tab_widget.widget(algorithm_idx).setCurrentIndex(result_idx)
```

Comment:
do we need the `result_idx` variable? we could set the index here to `len(results) - 1`

Reply:
assuming the `addTab` returns indices counting from 0